File: day-2/error-handlers-async.ts
export function isValid(value: number) {
    const maxValue = 10;
    const promise: Promise<string> = new Promise((resolve, reject) => {
        if (value <= maxValue) {
            resolve('valid')
        } else {
            reject('invalid')
        }
    })
    promise.then((result) => {
        console.log(`Your number is ${result}`);
    }).catch((result) => {
        console.log(`Your number is ${result}`);
    })
}

isValid(9);
isValid(10);
isValid(11);
isValid(12);

File: day-2/error-handlers-async.dart
import 'dart:async';

void isValid(int value) async {
  int maxValue = 10;
  Future<String> result;

  if (value <= maxValue) {
    result = Future.value('valid');
  } else {
    result = Future.error('invalid');
  }

  result
      .then((value) => {print('Your number is $value')})
      .catchError((error) => {print('Your number is $error')});
}

void main() {
  isValid(9);
  isValid(11);
}
